client/src/components/Profile.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import { Row, Col, Input, Button } from 'react-materialize';
+import { server_url } from '../config.json';
+
+
+export default class Profile extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            edit: false,
+            user: []
+        };
+
+        this.Avatar_image = this.Avatar_image.bind(this);
+        this.UpdateClick = this.UpdateClick.bind(this);
+        this.onChangeHandler = this.onChangeHandler.bind(this);
+        this.handleClick_Main = this.handleClick_Main.bind(this);
+        this.handleClick_Edit = this.handleClick_Edit.bind(this);
+    }
+
+    componentDidMount() {
+        fetch(server_url + '/profile/' + window.location.href.split("/").pop())
+        .then(response => response.json())
+        .then(json => {
+            this.setState({ user: json });
+        })
+
+    }
+
+    UpdateClick(eve) {
+        eve.preventDefault()
+        if (this.state.new_password !== null) {
+          if (this.state.new_password === this.state.repeat_password) {
+            const data = {
+                "email": this.state.email,
+                "first_name": this.state.f_name,
+                "last_name": this.state.l_name,
+                "nickname": this.state.nickname,
+                "location": this.state.location,
+                "password": this.state.new_password,
+                "id": this.state.user.id
+            }
+            fetch(server_url + '/profile/' + window.location.href.split("/").pop(), {
+                method: 'PUT',
+                headers: {'Content-Type':'application/json',
+                'Access-Control-Allow-Origin':'http://localhost:3000'},
+                body: JSON.stringify(data)
+            })
+            window.Materialize.toast("Profile Updated", 2500);
+          }
+          else {
+            window.Materialize.toast("Incorrect repeated password!", 2500);
+          }
+        }
+    }
+
+    onChangeHandler = (e) => {
+        let { id } = e.currentTarget
+        this.setState({ [id]: e.currentTarget.value })
+    }
+
+    handleClick_Main() {
+        this.setState({ edit: false });
+
+    }
+    handleClick_Edit() {
+        this.setState({ edit: true });
+    }
+
+    Avatar_image() {
+       if (this.state.user.avatar == null) {
+          return (
+            <img src="http://dialpharma.com/media/img/default_profile.png" alt="Default icon" />
+          );
+      }
+      else {
+         return (
+           <img src={this.state.user.avatar} alt="Avatar icon" />
+         );
+       }
+   }
+
+
+  render() {
+      if (this.state.edit === false) {
+          return (
+            <div>
+              <div>
+                <h1>Profile</h1>
+              </div>
+                <div id="main">
+                <Row>
+                    <Col s={6}>
+                        <Row>
+                            <this.Avatar_image />
```

Comment:
Seems strange, you can call it like a simple method {this.avatar_image}
Also better to name method smth like `renderAvatarImage`.
Better to use camelCase in JS [style code](https://github.com/airbnb/javascript/#naming--camelCase)